performance/memory-optimize-udtt.sql

```diff
@@ -71,9 +71,8 @@ declare @tt table(
 insert @tt(schema_id, t.name, user_type_id, table_type_object_id)
 select schema_id, t.name, user_type_id, type_table_object_id
 from sys.table_types t
-inner join sys.indexes i
-on t.type_table_object_id=i.object_id
-where is_user_defined=1 and is_memory_optimized=0 and i.name is not null
+where exists (select name, object_id from sys.indexes i where i.object_id=t.type_table_object_id and i.name is not null)

```

Comment:
```suggestion
where exists (select * from sys.indexes i where i.object_id=t.type_table_object_id and i.name is not null)

```

Reply:
Actually curious for future reference on the reasoning behind selecting * vs  only the columns I'm comparing, is it faster?

Reply:
Great question! Someone went into much detail on this exact topic here https://sqlquantumleap.com/2020/01/24/actual-difference-between-existsselect-1-existsselect-and-existsselect-column/. In a nutshell, both forms don't actually use the projection, and generate identical execution plans. The permissions edge case described in this blog post doesn't apply to our code, since we explicitly control our DB schema. We tend to prefer `exists (select * ...)` form for being easier on the eyes.